### PR TITLE
Make k8s clusters offline bootable

### LINKED ISSUE #69: option to pass custom provision file while (make cluster-up)
/kind enhancement

__Problem:__
 while working behind network proxy, `make cluster-up` fails to setup cluster while downloading CNI config. Also, no `docker pull` or `kubectl apply` is successful as no image gets downloaded due to lack of proxy settings in docker-daemon.

```
.
.
Your Kubernetes master has initialized successfully!

To start using your cluster, you need to run the following as a regular user:

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

You should now deploy a pod network to the cluster.
Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
  https://kubernetes.io/docs/concepts/cluster-administration/addons/

You can now join any number of machines by running the following on each node
as root:

  kubeadm join 192.168.66.101:6443 --token abcdef.1234567890123456 --discovery-token-ca-cert-hash sha256:23d5e3383df6878d799c4b326f8910d521894dc9a403dcac9dd6a3800236fac3

+ kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
The connection to the server raw.githubusercontent.com was refused - did you specify the right host or port?
provisioning node node01 failed
Makefile:104: recipe for target 'cluster-up' failed
make: *** [cluster-up] Error 1
```
__Expected Behaviour__
- modify provision scripts like `node.sh` or `node<node-no>.sh` etc.
- `gocli run ` should be able to consume custom scripts if indicated explicitly

### PR DESCRIPTION
**Pull request by rmohr**
Don't pull cni manifests from the internet on each cluster start.

Fixes #69

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
What regarding publish jobs and updating the README?

**Comment by rmohr** [comment]
Seems like I first have to find out what changed in centos so that  1.10.11 and 1.11.0 can build again.

**Comment by cynepco3hahue** [comment]
@rmohr hm, maybe `kubernetes-cni-0.6.0` does not suit well for all versions of the k8s?

**Comment by rmohr** [comment]
They already failed before I switched the CNI version.

**Comment by cynepco3hahue** [comment]
@rmohr wasn't aware of it :worried:

File: k8s/scripts/provision.sh [hunk 1]
```diff
@@ -94,10 +94,11 @@ echo br_netfilter >> /etc/modules
 
 kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version v${version} --token abcdef.1234567890123456
 if [[ ${BASH_REMATCH[1]} -ge "12" ]]; then
-kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/a70459be0084506e4ec919aa1c114638878db11b/Documentation/kube-flannel.yml
+    curl https://raw.githubusercontent.com/coreos/flannel/a70459be0084506e4ec919aa1c114638878db11b/Documentation/kube-flannel.yml --output /etc/kubernetes/cni.yml
```
Comment (SchSeba) [inline k8s/scripts/provision.sh]: can you maybe download the file and put it under the manifest folder ?

Comment (rmohr) [reply k8s/scripts/provision.sh]: done.

**Comment by rmohr** [comment]
/approve

@all everything should be fine. Waiting for a final review and hopefully lgtm.

**Pull request merged**